src/api/worker/offline/OfflineStorageMigrator.ts

```diff
@@ -105,4 +110,33 @@ export class OfflineStorageMigrator {
 			await storage.setStoredModelVersion(app, version)
 		}
 	}
+
+	/**
+	 * it's possible that the user installed an older client over a newer one, and we don't have backwards migrations.
+	 * in that case, it's likely that the client can't even understand the contents of the db.
+	 * we're going to delete it and not migrate at all.
+	 * @private
+	 */
+	private isNewerThanDatabase(meta: Partial<OfflineDbMeta>): boolean {
```

Comment:
I would invert the function and probably rename it to something like `isDbNewerThanCurrentClient()` or even `isDowngrade()`. I  understand the idea that `meta` that we pass is never than db but it took me a few minutes to figure out what return value means.